pkg/resources/provisioning.cattle.io/v1/cluster/validator.go

```diff
@@ -184,6 +188,29 @@ func (p *provisioningAdmitter) validateClusterName(request *admission.Request, r
 	return nil
 }
 
+func (p *provisioningAdmitter) validateMachinePoolNames(request *admission.Request, response *admissionv1.AdmissionResponse, cluster *v1.Cluster) error {
+	if request.Operation != admissionv1.Create {
+		return nil
+	}
+
+	if cluster.Spec.RKEConfig == nil {
+		return nil
+	}
+
+	for _, pool := range cluster.Spec.RKEConfig.MachinePools {
+		if len(pool.Name) > 63 {
+			response.Result = &metav1.Status{
```

Comment:
You might want to `break` after finding the first long pool name? There is no need to keep looping...